.github/workflows/build_test_package.yml

```diff
@@ -2,740 +2,320 @@ name: Build, Test and Package
-          java-version: '21'
-
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v3
-
+          distribution: "temurin"
+          java-version: "21"
       - name: Run unit tests
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew test -x spotlessCheck -x openApiValidate
-
-  connector_it:
+        run: |
+          ./gradlew test -x spotlessCheck -x openApiValidate
+  connector-it:
     runs-on: ubuntu-latest
-    needs:
-      - test
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
+    needs:
+      - test-unit
     permissions: read-all
-
-    if: github.event_name != 'pull_request_target'
-
     steps:
       - uses: actions/checkout@v4
-
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
-          distribution: 'temurin'
-          java-version: '21'
-
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v3
-
+          distribution: "temurin"
+          java-version: "21"
       - name: Run Connector integration tests
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew :cosmotech-connector-api:integrationTest -x spotlessCheck -x openApiValidate
-
-  dataset_it:
+        run: |
+          ./gradlew :cosmotech-connector-api:integrationTest -x spotlessCheck -x openApiValidate
+  dataset-it:
     runs-on: ubuntu-latest
-    needs:
-      - test
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
+    needs:
+      - test-unit
     permissions: read-all
-
-    if: github.event_name != 'pull_request_target'
-
     steps:
       - uses: actions/checkout@v4
-
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
-          distribution: 'temurin'
-          java-version: '21'
-
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v3
-
+          distribution: "temurin"
+          java-version: "21"
       - name: Run RBAC Dataset integration tests
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew :cosmotech-dataset-api:integrationTest --tests "com.cosmotech.dataset.service.DatasetServiceRBACTest" -x spotlessCheck -x openApiValidate
-
+        run: |
+          ./gradlew :cosmotech-dataset-api:integrationTest \
+            --tests "com.cosmotech.dataset.service.DatasetServiceRBACTest" \
+            -x spotlessCheck \
+            -x openApiValidate
       - name: Run Dataset integration tests
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew :cosmotech-dataset-api:integrationTest --tests "com.cosmotech.dataset.service.DatasetServiceIntegrationTest" -x spotlessCheck -x openApiValidate
-
-  organization_it:
+        run: |
+          ./gradlew :cosmotech-dataset-api:integrationTest \
+            --tests "com.cosmotech.dataset.service.DatasetServiceIntegrationTest" \
+            -x spotlessCheck \
+            -x openApiValidate
```

Comment:
If you want to span the command on multiple line I'd suggest using the `>` multiline YAML marker instead of `|`, this way you also remove the ` \` line breaks.